### PR TITLE
qa/cephtool: remove/deprecate proxy cache-tier mode

### PR DESCRIPTION
**Pull request by smithfarm**
NOTE: to be backported via https://tracker.ceph.com/issues/45514

---

Fixes: https://tracker.ceph.com/issues/45614
Fixes: a0a3ed324a130a9934735927600f01d1fa65bc0d
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
**running** http://pulpito.ceph.com/smithfarm-2020-05-19_20:20:02-rados:singleton-bluestore-master-distro-basic-smithi/

**Comment by smithfarm** [comment]
**running** http://pulpito.ceph.com/smithfarm-2020-05-19_20:46:19-rados:singleton-bluestore-master-distro-basic-smithi/

File: qa/workunits/cephtool/test.sh [hunk 1]
```diff
@@ -351,14 +351,16 @@ function test_tiering_1()
   expect_false ceph osd tier cache-mode cache forward
   expect_false ceph osd tier cache-mode cache forward --yes-i-really-mean-it
   expect_false ceph osd tier cache-mode cache readonly
-  ceph osd tier cache-mode cache proxy
+  # proxy is removed/deprecated
+  expect_false ceph osd tier cache-mode cache proxy
```
Comment (tchaikov) [inline qa/workunits/cephtool/test.sh:355]: could you drop this line? no need to test an unsupported cache mode.

File: qa/workunits/cephtool/test.sh [hunk 2]
```diff
@@ -351,14 +351,16 @@ function test_tiering_1()
...
+  ceph osd tier cache-mode cache readproxy
   ceph osd tier cache-mode cache none
   ceph osd tier cache-mode cache readonly --yes-i-really-mean-it
   expect_false ceph osd tier cache-mode cache forward
   expect_false ceph osd tier cache-mode cache forward --yes-i-really-mean-it
   ceph osd tier cache-mode cache none
   ceph osd tier cache-mode cache writeback
-  ceph osd tier cache-mode cache proxy
+  expect_false ceph osd tier cache-mode cache proxy
```
Comment (tchaikov) [inline qa/workunits/cephtool/test.sh:363]: ditto.

File: qa/workunits/cephtool/test.sh (hunk 1 above)
Comment (smithfarm) [reply qa/workunits/cephtool/test.sh:355]: this line is testing that `ceph osd tier cache-mode cache proxy` really fails as expected, similar to the previous three lines which test the same for `forward` and `readonly` caching modes.

File: qa/workunits/cephtool/test.sh (hunk 2 above)
Comment (smithfarm) [reply qa/workunits/cephtool/test.sh:363]: removed. Thanks!

**Comment by smithfarm** [comment]
**running** http://pulpito.ceph.com/smithfarm-2020-05-20_16:08:46-rados:singleton-bluestore-wip-proxy-is-removed-distro-basic-smithi/

**Comment by smithfarm** [comment]
**pass** http://pulpito.ceph.com/smithfarm-2020-05-20_20:41:34-rados:singleton-bluestore-wip-proxy-is-removed-distro-basic-smithi/

**Comment by smithfarm** [comment]
@neha-ojha @tchaikov @jdurgin This fixes the test broken by https://github.com/ceph/ceph/commit/a0a3ed324a130a9934735927600f01d1fa65bc0d

File: qa/workunits/cephtool/test.sh [hunk 3]
```diff
@@ -345,29 +345,27 @@ function test_tiering_1()
...
   ceph osd tier cache-mode cache none
   ceph osd tier cache-mode cache readonly --yes-i-really-mean-it
-  expect_false ceph osd tier cache-mode cache forward
-  expect_false ceph osd tier cache-mode cache forward --yes-i-really-mean-it
   ceph osd tier cache-mode cache none
   ceph osd tier cache-mode cache writeback
-  ceph osd tier cache-mode cache proxy
-  ceph osd tier cache-mode cache writeback
   expect_false ceph osd tier cache-mode cache none
-  expect_false ceph osd tier cache-mode cache readonly --yes-i-really-mean-it
```
Comment (smithfarm) [inline qa/workunits/cephtool/test.sh:364]: this line should not be deleted I guess

**Comment by smithfarm** [comment]
**running** http://pulpito.ceph.com/smithfarm-2020-05-21_11:42:36-rados:singleton-bluestore-wip-proxy-is-removed-distro-basic-smithi/

**Comment by neha-ojha** [comment]
No cephtool related failures in rados run: http://pulpito.ceph.com/nojha-2020-05-21_02:01:24-rados-wip-35139-distro-basic-smithi/

**Pull request merged**